test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewResultTest.cs

```diff
@@ -97,6 +97,54 @@ public class ViewResultTest
             viewEngine.Verify();
         }
 
+        [Fact]
+        public async Task ExecuteResultAsync_ViewResultAllowNullViewDataAndTempData()
+        {
+            var context = GetActionContext();
+            var controller = new TestController(context);
+
+            await controller.Index().ExecuteResultAsync(context);
+            Assert.Null(controller.ViewContext.ViewData);
+            Assert.Null(controller.ViewContext.TempData);
+        }
+
+        private class TestController
```

Comment:
Ah that's actually a good point. Can just new up the type and call Execute from within the test.